packages/devtools_app/lib/src/shared/console/eval/eval_service.dart

```diff
@@ -177,12 +177,23 @@ class EvalService extends DisposableController with AutoDisposeControllerMixin {
   }
 
   Future<Map<String, String>?> _scopeIfSupported(String isolateRefId) async {
-    // Debugging for web does not support scopes yet.
-    if (await serviceManager.connectedApp?.isDartWebApp ?? true) return null;
+    if (messageIfScopeNotSupported() != null) return null;
 
     return scope.value(isolateId: isolateRefId);
   }
 
+  /// If scope is supported, returns null.
+  ///
+  /// Otherwise, returns a message explaining why scope is not supported.
+  String? messageIfScopeNotSupported() {
+    // Web does not support scopes yet.
+    final isWeb = serviceManager.connectedApp?.isDartWebAppNow ?? true;
+    if (isWeb) {
+      return 'Scope variables are not supported for web applications.';
```

Comment:
instead of returning a string, wdyt about emitting to the console from this method, and then returning a bool indicating whether or not scope is supported. Then we could just check the bool value returned `isScopeSupported` instead of checking the nullness of a string.

Reply:
we could add a parameter `emitWarningToConsole` if there are cases where we want to perform the check but not emit a message to console.

Reply:
+1, if we're only using this message in one spot at the moment, it'd be cleaner to have this method return a boolean and then print this message when that check returns false.